.github/workflows/release.yml

```
@@ -48,5 +48,5 @@ jobs:
                   NUGET_SOURCE: ${{ secrets.NUGET_SOURCE }}
                   RELEASE_USER_GITHUB: ${{ secrets.RELEASE_USER_GITHUB }}
                   REPOSITORY_NAME_GITHUB: ${{ secrets.REPOSITORY_NAME_GITHUB }}
-                  TOKEN_GITHUB: ${{ secrets.TOKEN_GITHUB }}
+                  GITHUB_TOKEN: ${{ secrets.TOKEN_GITHUB }}
```

Actually it should have been like this. Maybe at the time of pipeline `GITHUB_TOKEN` not existed.
```suggestion
                  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Probably we can even remove this line

---

It works like this here
https://github.com/fsprojects/FSharp.Data.GraphQL/blob/d96b065b95d553cdaac68cd2697490c5b618be44/.github/workflows/publish_ci.yml#L119

---

Can the token passed from environment be more restrictive then the one set manually?

---

It can be, but even the token permissions can be described in the yaml, so it should be more clear what's required. See https://docs.github.com/en/actions/security-for-github-actions/security-guides/automatic-token-authentication#modifying-the-permissions-for-the-github_token for details.